encord/client.py

```diff
@@ -1304,24 +1384,30 @@ def model_train(
                 message="You must pass weights from the `encord.constants.model_weights` module to train a model."
             )
 
-        training_params = ModelTrainingParams(
-            {
-                "label_rows": label_rows,
-                "epochs": epochs,
-                "batch_size": batch_size,
-                "weights": weights,
-                "device": _device_to_string(device),
-            }
+        training_hash = self.__model_train_start(
+            model_hash=uid,
+            label_rows=label_rows,
+            epochs=epochs,
+            batch_size=batch_size,
+            model=weights.model,
+            training_weights_link=weights.training_weights_link,
+            device=_device_to_string(device),
         )
 
-        model = Model(
-            {
-                "model_operation": ModelOperations.TRAIN.value,
-                "model_parameters": training_params,
-            }
+        res = self.__model_train_get_result(
```

Comment:
Let's not do the polling here. Instead - return a token and leave it to the caller to poll on it - only if they want to